### ISSUE TITLE
Error in solve.default

### ISSUE THREAD

**Issue by JulesText** [grf-labs/grf#610]
There seems to be an unexpected error in rare cases with the average_partial_effect function in version 1.0.1

The forest object producing the error is at [this link](https://github.com/JulesText/pub/blob/master/bug_average_partial_effect.forest.RData).

Loading the object, then running the following code produces the error.

```
pred <- predict(f, estimate.variance = TRUE)

average_partial_effect(f)

tau.hat <- pred[['predictions']]
tau.hat.hi <- tau.hat > median(tau.hat)
average_partial_effect(f, subset = tau.hat.hi)
average_partial_effect(f, subset = !tau.hat.hi)

```

The error is:
```
Error in solve.default(t(A) %*% A, bias) : 
  system is computationally singular: reciprocal condition number = 3.07477e-17

```
It seems that the line in average_partial_effect generating the error is:

`lambda <- solve(t(A) %*% A, bias)`

My guess is that the matrix this is trying to solve is not invertible because its determinant is zero. Calculating the determinant shows it is near zero.

```
det(t(A) %*% A, bias)
[1] -2.872738e-20
```

Maybe adding an error catch for when the determinant is near zero, with exception handling, could prevent the error message. 

As an aside, changing the code in average_partial_effect from
```
subset.weights.raw <- observation.weight[subset]
subset.weights <- subset.weights.raw / mean(subset.weights.raw)
```
to as it was in version 0.10.3
```
subset.weights <- observation.weight[subset]
```
avoids the error, in this example, though this appears coincidental to me. Here, the determinant is slightly further from zero.
```
det(t(A) %*% A, bias)
[1] -4.03068e-19
```

**Comment by swager** [comment]
@JulesText thanks for the very detailed bug report! The referenced patch to the `average_partial_effect` function fixes the issue, at least on my machine. Let me know if the issue still persists for you.

**Comment by erikcs** [comment]
Closing as should be resolved in #611

**Comment by JulesText** [comment]
Apology for the delayed response, I appreciate you patching the issue @swager. The exception you added fixes the issue.
